core/src/main/java/org/mule/runtime/core/api/management/stats/CursorComponentDecoratorFactory.java

```diff
@@ -67,6 +68,18 @@
    */
   InputStream decorateInput(InputStream decorated, String correlationId);
 
+  /**
+   * If statistics are enabled, decorates the provided {@link CursorStream}.
+   * <p>
+   * Ref: {@link PayloadStatistics#getInputByteCount()}.
+   *
+   * @param decorated the {@link CursorStream} to decorate.
+   * @param correlationId information to be used in the case a detailed report needs to be obtained, allowing to match the
+   *        measured volume to a specific execution.
+   * @return the decorated {@link CursorStream}.
+   */
+  CursorStream decorateInput(CursorStream decorated, String correlationId);
```

Comment:
why is this case different to the one with `InputSream`?

Reply:
When I want to decorate a CursorProvider on openCursor I need a CursorStream as a result. PayloadStatisticsInputStream extends InputStream, not CursorStream.
This is the wrapper for CursorStreamProvider which returns decorated cursors:
https://github.com/mulesoft/mule/pull/9338/files#diff-90fb1d0f6ef60883cd38dbf9b5bc8161R32